File: java_benchmarks/src/main/java/com/cmpe220/benchmark/MongoSelect100.java
package com.cmpe220.benchmark;

import java.util.ArrayList;
import java.util.Iterator;

import com.mongodb.*;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;

import org.bson.Document;

import com.mongodb.client.FindIterable;
import com.mongodb.client.MongoClient;

/**
* REQUIRED: have Cassandra DB already running
*/
public class MongoSelect100 extends AbstractBenchmark {

    private MongoClient client;
    private MongoDatabase db;
    private MongoCollection<Document> col;
    private ArrayList<Document> data;
    private ArrayList<Document> x;

    public MongoSelect100() {
        category = "mongo";
        description = "This is an example";
    }

    public void setupQuery() {
        System.out.println("Connecting to mongo...");
        client = MongoClients.create("mongodb://localhost:27017/");
        db = client.getDatabase("mydb");
        String colName = "mycol";
        db.createCollection(colName);
        col = db.getCollection(colName);
        data = new ArrayList<Document>();
        for (int i=0; i<100; i++) {
            data.add(
                new Document("fname", "Rutuja")
                    .append("lname", "Palatkar")
                    .append("school", "SJSU")
                    .append("role", "Student")
                    .append("id", "ID")
            );
        }
        col.insertMany(data);
        x = new ArrayList<Document>();
    }

    public void endQuery() {
        System.out.println("Closing...");
        col.drop();
        client.close();
    }

    public void runQuery() {
        FindIterable<Document> iterObj = col.find();
        Iterator<Document> i = iterObj.iterator();
        while (i.hasNext()) {
            x.add(i.next());
        }
    }
}

File: python_benchmarks/Mongo_Select100.py
'''
REQUIRED: have Cassandra DB already running
pip install cassandra-driver (https://docs.datastax.com/en/developer/python-driver/3.25/)

(This is run from the parent directory, so
"python_benchmarks.AbstractBenchmark"
is required instead of "AbstractBenchmark")
'''
from python_benchmarks.AbstractBenchmark import AbstractBenchmark
import pymongo


class Mongo_Select100(AbstractBenchmark):

    def __init__(self):
        self.category = 'mongo'
        self.description = 'This is an example'

    def setupQuery(self):
        print('Connecting to mongo...')
        self.client = pymongo.MongoClient('mongodb://localhost:27017/')
        self.db = self.client['mydb']
        self.col = self.db['mycol']
        self.col.insert_many([{
            'fname': 'Rutuja',
            'lname': 'Palatkar',
            'school': 'SJSU',
            'role': 'Student',
            'id': 'ID',
        } for _ in range(100)])

    def endQuery(self):
        self.col.drop()
        self.client.close()

    def runQuery(self):
        x = [y for y in self.col.find()]
